sqlx-core/src/types/json.rs

```diff
@@ -1,3 +1,24 @@
+//! Json for json and jsonb fields
+//! 
+//! Will attempt to cast to type passed in as the generic.
+//! ```
+//! #[derive(FromRow)]
+//! struct Author {
+//!   name: String,
+//!   books: types::Json<Book>
+//! }
+//! ```
+//! 
+//! Can also be used to turn the json/jsonb into a hashmap
+//! ```
+//! #[derive(FromRow)]
+//! struct FooStruct {
+//!   id: String,
+//!   properties: types::Json<HashMap<String, BarStruct >>
+//! }
+//! ```
+//! 
+
```

Comment:
These examples won't be rendered as the `json` module isn't public. You should move these to the `Json` type itself.

Reply:
Doc examples are tested as a separate crate so they also need to include imports for any non-local types.

Reply:
Thanks for the feedback, learned a bunch